### DISCUSSION TITLE
Declarative Language Support for Multi-Command Execution

### DISCUSSION THREAD

**Discussion by infamousjoeg** [infamousjoeg/cybr-cli#112] [Ideas]
This idea was submitted by @CaptainFluffyToes.

He would like a declarative language, such as YAML, for executing multiple commands instead of having to create a Shell wrapper.  The manifest would be loaded, similar to Kubectl, and it would execute the multiple commands in a "run book style" similar to Ansible.

I asked if we could use CSV to support bulk data upload, such as Safe names for creation or Account metadata for onboarding.  Awaiting his response here!

**Comment by AndrewCopeland** [comment]
I don't like this approach as it feels like it will end up being a DSL, which I do not want to maintain or support.

**Comment by infamousjoeg** [comment]
Well that was a short lived discussion. lol
